regolith/helpers/l_contactshelper.py

```diff
@@ -0,0 +1,131 @@
+    """
+    # btype must be the same as helper target in helper.py
+    btype = HELPER_TARGET
+    needed_dbs = [f'{TARGET_COLL}']
+
+    def construct_global_ctx(self):
+        """Constructs the global context"""
+        super().construct_global_ctx()
+        gtx = self.gtx
+        rc = self.rc
+        if "groups" in self.needed_dbs:
+            rc.pi_id = get_pi_id(rc)
+        rc.coll = f"{TARGET_COLL}"
+        try:
+            if not rc.database:
+                rc.database = rc.databases[0]["name"]
+        except:
+            pass
+        colls = [
+            sorted(
+                all_docs_from_collection(rc.client, collname), key=_id_key
+            )
+            for collname in self.needed_dbs
+        ]
+        for db, coll in zip(self.needed_dbs, colls):
+            gtx[db] = coll
+        gtx["all_docs_from_collection"] = all_docs_from_collection
+        gtx["float"] = float
+        gtx["str"] = str
+        gtx["zip"] = zip
+
+
+    def sout(self):
+        rc = self.rc
+        contacts = []
+        if rc.date:
+            temp_dat = date_parser.parse(rc.date).date()
+        for contact in self.gtx["contacts"]:
+            if rc.name:
+                list_add = False
+                for nam in rc.name:
+                     if nam.casefold() in contact.get('name').casefold():
+                        list_add = True
+                        break
+                if not list_add:
+                    continue
+            if rc.inst:
+                if not (rc.inst.casefold() in contact.get('institution').casefold()):
+                    continue
+            if rc.date:
+                if rc.range:
+                    temp_dict = {"begin_date":temp_dat - dateutil.relativedelta.relativedelta(months=int(rc.range)), "end_date":temp_dat + dateutil.relativedelta.relativedelta(months=int(rc.range))} 
+                else:
+                    temp_dict = {"begin_date":temp_dat - dateutil.relativedelta.relativedelta(months=2), "end_date":temp_dat + dateutil.relativedelta.relativedelta(months=2)}                
+                if not is_current(temp_dict, now=temp_dat):
+                    continue
+            if rc.notes:
+                list_add = False
+                for note in rc.notes:
+                    if isinstance(contact.get('notes'),str):
+                        if note.casefold() in contact.get('notes').casefold():
+                            list_add = True
+                    elif isinstance(contact.get('notes'),list):
+                        for no in contact.get('notes'):
+                             if note.casefold() in no.casefold():
+                                 list_add = True
+                if not list_add:
+                    continue
+            contacts.append(contact)
+        for con in contacts:
+            if con.get('email'):
+                print(f"name: {con.get('name')}, institution: {con.get('institution')}, email: {con.get('email','missing')}")
```

Comment:
The new is_current still doesn't fix the issue I had earlier. My date dictionary uses datetime objects. is_current wants a dictionary with string objects but it converts those strings to datetime objects in the function. So, me using the existing is_current would be double the work, as I would have to convert my datetime objects to strings just to use is_current but those strings would be converted back to datetime objects in the function. The reason I copied the function and modified it was to avoid the intermediate conversion.